#                                            $$ Titanic$$$$ Case$$$$ Study $$

In this case study we need to identify the probability of variaous dependent variables in dataset with respect to other independent variable

In [88]:
#importing important libraries
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
%matplotlib inline
import seaborn as sns

# Loading titanic dataset

In [89]:
dataset=pd.read_csv("train.csv")

In [90]:
dataset.head()

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4      0            373450   8.0500   NaN        S

In [91]:
dataset.shape

(889, 12)

# Checking for null values in dataset

In [92]:
dataset.isnull().sum()

PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
Age              0
SibSp            0
Parch            0
Ticket           0
Fare             0
Cabin          687
Embarked         0
dtype: int64

### more no. of null values pressent in column Cabin 
### therefore,, we can drop this entire column

In [93]:
del dataset["Cabin"]

In [94]:
dataset.head(1)

PassengerId  Survived  Pclass                     Name   Sex   Age  SibSp  \
0            1         0       3  Braund, Mr. Owen Harris  male  22.0      1   

   Parch     Ticket  Fare Embarked  
0      0  A/5 21171  7.25        S

### Now drop unwanted columns that are irrelevant 

In [95]:
dataset.drop(["PassengerId","Name","Ticket"],axis=1,inplace=True)

In [96]:
dataset.head(1)

Survived  Pclass   Sex   Age  SibSp  Parch  Fare Embarked
0         0       3  male  22.0      1      0  7.25        S

In [97]:
dataset.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 889 entries, 0 to 888
Data columns (total 8 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   Survived  889 non-null    int64  
 1   Pclass    889 non-null    int64  
 2   Sex       889 non-null    object 
 3   Age       889 non-null    float64
 4   SibSp     889 non-null    int64  
 5   Parch     889 non-null    int64  
 6   Fare      889 non-null    float64
 7   Embarked  889 non-null    object 
dtypes: float64(2), int64(4), object(2)
memory usage: 55.7+ KB


In [98]:
from sklearn.preprocessing import LabelEncoder
lab=LabelEncoder()

dataset.Sex=lab.fit_transform(dataset.Sex)
dataset.Embarked=lab.fit_transform(dataset.Embarked)

In [99]:
dataset.head()

Survived  Pclass  Sex   Age  SibSp  Parch     Fare  Embarked
0         0       3    1  22.0      1      0   7.2500         2
1         1       1    0  38.0      1      0  71.2833         0
2         1       3    0  26.0      0      0   7.9250         2
3         1       1    0  35.0      1      0  53.1000         2
4         0       3    1  35.0      0      0   8.0500         2

In [100]:
dataset.columns

Index(['Survived', 'Pclass', 'Sex', 'Age', 'SibSp', 'Parch', 'Fare',
       'Embarked'],
      dtype='object')

# Naive Bayes

In [101]:
from sklearn.model_selection import train_test_split
from sklearn.naive_bayes import*
from sklearn.metrics import accuracy_score,confusion_matrix

In [102]:
def NB(col):
    clf=BernoulliNB()
    
    for i in col:
    
        x=dataset.drop([i],axis=1)
        print("'{}' is dependent variable and remaing are independent variables".format(i),end="\n\n")
        print("x is :",x.head(3))
        y=dataset[i]
        print("y is ;",y.head(3))
        x_train,x_test,y_train,y_test=train_test_split(x,y,test_size=0.3,random_state=0)
        ypred=clf.fit(x_train,y_train).predict(x_test)
        print("accuracy score for '{0}'' is : {1}  ".format(i,accuracy_score(y_test,ypred,normalize=True)),end="\n\n")
        print("confusion matrix for '{0}' is: \n {1}".format(i,confusion_matrix(y_test,ypred)),end="\n\n\n")
    

In [103]:
NB(['Survived', 'Pclass', 'Sex', 'SibSp', 'Parch','Embarked']) 

'Survived' is dependent variable and remaing are independent variables

x is :    Pclass  Sex   Age  SibSp  Parch     Fare  Embarked
0       3    1  22.0      1      0   7.2500         2
1       1    0  38.0      1      0  71.2833         0
2       3    0  26.0      0      0   7.9250         2
y is ; 0    0
1    1
2    1
Name: Survived, dtype: int64
accuracy score for 'Survived'' is : 0.7715355805243446  

confusion matrix for 'Survived' is: 
 [[131  26]
 [ 35  75]]


'Pclass' is dependent variable and remaing are independent variables

x is :    Survived  Sex   Age  SibSp  Parch     Fare  Embarked
0         0    1  22.0      1      0   7.2500         2
1         1    0  38.0      1      0  71.2833         0
2         1    0  26.0      0      0   7.9250         2
y is ; 0    3
1    1
2    3
Name: Pclass, dtype: int64
accuracy score for 'Pclass'' is : 0.5917602996254682  

confusion matrix for 'Pclass' is: 
 [[ 30   8  32]
 [  9  10  30]
 [ 24   6 118]]


'Sex' is dependent variable and

# Conclusion:
## for dependendent variable "Survived":
1. accuracy score is 0.7715 i.e. 77% model is accurate.
2. Out of 267 peoples 206 peoples data is correctly predicted while remaing are not predicted properly

## for dependent variable "Pclass":
1. accuracy score is 0.591760299625468 i.e. 60%  model is accurate.
2. Out of 267 peoples 158 peoples data is correctly predicted while remaing are not predicted properly

## for dependent variable "Sex ":
1. accuracy score is 0.7453183520599251 i.e. 74%  model is accurate.
2. Out of 267 peoples 199 peoples data is correctly predicted while remaing are not predicted properly

## for dependent variable "SibSp":
1. accuracy score is  0.7153558052434457 i.e. 71%  model is accurate.
2. Out of 267 peoples 184 peoples data is correctly predicted while remaing are not predicted properly

## for dependent variable "Parch" :
1. accuracy score is  0.6891385767790262 i.e. 69%  model is accurate.
2. Out of 267 peoples 191 peoples data is correctly predicted while remaing are not predicted properly


## for dependent variable "Embarked" :
1. accuracy score is  0.7340823970037453  i.e. 73%  model is accurate.
2. Out of 267 peoples 196 peoples data is correctly predicted while remaing are not predicted properly



